RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegFit.h

```diff
@@ -0,0 +1,190 @@
+      std::stringstream gemetapartmapss; gemetapartmapss<<"[GEMCSCSegFit::ctor] :: gemetapartmap :: elements ["<<std::endl;
+      for(std::map<uint32_t, const GEMEtaPartition*>::const_iterator mapIt = gemetapartmap_.begin(); mapIt != gemetapartmap_.end(); ++mapIt)
+	{
+	  gemetapartmapss<<"[GEM DetId "<<mapIt->first<<" ="<<GEMDetId(mapIt->first)<<", GEM EtaPart "<<mapIt->second<<"],"<<std::endl;
+	}
+      gemetapartmapss<<"]"<<std::endl;
+      std::string gemetapartmapstr = gemetapartmapss.str();
+      edm::LogVerbatim("GEMCSCSegFit") << gemetapartmapstr;
+      // --- End LogDebug -----------------------------------------------------------------------------------
+    } 
+
+  virtual ~GEMCSCSegFit() {}
+
+  // Least-squares fit
+  void fit( void ); // fill uslope_, vslope_, intercept_  @@ FKA fitSlopes()
+  // Calculate covariance matrix of fitted parameters
+  AlgebraicSymMatrix covarianceMatrix(void);
+
+  // Change scale factor of rechit x error 
+  // - expert use only!
+  void setScaleXError ( double factor ) { scaleXError_ = factor; }
+
+  // Fit values
+  float xfit( float z ) const;
+  float yfit( float z ) const;
+
+  // Deviations from fit for given input (local w.r.t. chamber)
+  float xdev( float x, float z ) const;
+  float ydev ( float y, float z ) const;
+  float Rdev( float x, float y, float z ) const;
+
+  // Other public functions are accessors
+  std::vector<const TrackingRecHit*> hits(void) const { return hits_; }
+  double scaleXError(void) const { return scaleXError_; }
+  size_t nhits(void) const { return hits_.size(); }
+  double chi2(void) const { return chi2_; }
+  int ndof(void) const { return ndof_; }
+  LocalPoint intercept() const { return intercept_;}
+  LocalVector localdir() const { return localdir_;}
+
+  const CSCChamber*      cscchamber     (uint32_t id) const { 
+    if(csclayermap_.find(id)==csclayermap_.end()) { edm::LogVerbatim("GEMCSCSegFit") << "[GEMCSCSegFit] Failed to find CSCChamber in CSCLayerMap"; return 0;}
+    else { return (csclayermap_.find(id)->second)->chamber(); } 
+  }
+  const CSCLayer*        csclayer       (uint32_t id) const { 
+    if(csclayermap_.find(id)==csclayermap_.end()) { edm::LogVerbatim("GEMCSCSegFit") << "[GEMCSCSegFit] Failed to find CSCLayer in CSCLayerMap"; return 0;}
+    else { return csclayermap_.find(id)->second; }
+  }
+  const GEMEtaPartition* gemetapartition(uint32_t id) const { 
+    if(gemetapartmap_.find(id)==gemetapartmap_.end()) { edm::LogVerbatim("GEMCSCSegFit") << "[GEMCSCSegFit] Failed to find GEMEtaPartition in GEMEtaPartMap"; return 0;}
+    else { return gemetapartmap_.find(id)->second; }
+  }
+  const CSCChamber*      refcscchamber  ()            const { 
+    if(csclayermap_.find(refid_)==csclayermap_.end()) { edm::LogVerbatim("GEMCSCSegFit") << "[GEMCSCSegFit] Failed to find Reference CSCChamber in CSCLayerMap"; return 0;}
+    else { return (csclayermap_.find(refid_)->second)->chamber(); }
+  }
```

Comment:
The only thing i can do here is to throw an exception in case no valid pointer is found.